src/components/tabs/tabs.ts

```diff
@@ -79,6 +88,19 @@ export class MdTabGroup {
     return this._selectedIndex;
   }
 
+  /**
+   * Determines if an index is valid.  If the tabs are not ready yet, we assume that the user is
+   * providing a valid index and return true.
+   */
+  isValidIndex(index: number): boolean {
+    if (this._tabs) {
+      const tab = this._tabs.toArray()[index];
+      return tab && !tab.disabled;
+    } else {
+      return true;
```

Comment:
Ah, I should add a comment explaining why.  When the component is first loaded, the tabs aren't ready yet, so we allow any value to pass through.  In the case of async tabs, we don't know when, exactly, the tabs will be ready, so I handle it in a generic way that does nothing while the component is in an invalid state (ie. no tabs).